src/main/java/core/AbstractPlayer.java

```diff
@@ -32,6 +32,9 @@ public final int getPlayerID() {
     public final AbstractForwardModel getForwardModel() {
         return forwardModel;
     }
+    public void setForwardModel(AbstractForwardModel model) {
```

Comment:
I don't like the idea of this, having it public would break the confidence that the FM received is the proper one for the game played, as set in the Game class. I could see some uses for it being public, is there something specific in this PR that needs it?

Reply:
The specific reason in this PR is in MCTSPlayer when it contains distinct rollout strategies or opponent models that  themselves are implementations of AbstractPlayer (so can be much more sophisticated than random). These therefore also need to be initialised with the forward model by Game...but are nested within MCTSPlayer, so the setForwardModel() method here is recursive, with MCTSPlayer forwarding on the model to its internal components when it is received from Game.

One option I toyed with was including this within initializePlayer(), to be passed at the same time as the initial game state; but that is also public so not an improvement.

Another option might be to have Game pass ForwardModel at the same time as AbstractGameState when it calls getAction()? (although again, this is a public method...)

(Also - I think it might be quite reasonable for an AbstractPlayer to use a non-perfect forward model.)

Reply:
I'm more worried of a case where a player is accidentally given a FM for a different game through this, which would make all sorts of weird things happen, some not obvious - and if we open this in competition format, it'd also raise the question of can any player hack another player's FM and break the confidence we'd have in fairness there? But I can see potential for the other use cases you've mentioned, maybe add a warning in docs / an issue for us to check in future competition setting?

Reply:
I'm much less worried about this. This fake forward model 'attack' would require the other player to get a reference to our AbstractPlayer, and I don't see how this is possible, as this is held within the Game, and the at no point is a reference to the Game passed to the players - they get the AbstractGameState only, which also does not have any reference to the Game.
...am I missing something?

Reply:
Right, at the moment this is not the case, but with the framework expanding and more things becoming public - and there being more and more things, easy to start to lose track of details -, I can see this becoming an issue. Adding a note to the docs would give piece of mind.

Reply:
Ok - added some notes on this.

For competitions in the future we could perhaps pass the ForwardModel to the AbstractPlayer along with the AbstractGameState in getAction(). That way any 'corrupted' agent could always refresh its copy.

Reply:
True, that's a good option! Thanks! :) Approved for merge!